tests/tests_integration/test_api/test_data_workflows.py

```diff
@@ -291,138 +282,115 @@ def workflow_data_modeling_trigger(cognite_client: CogniteClient, add_multiply_w
+        listed_by_wf_version_id = cognite_client.workflows.versions.list(
+            WorkflowVersionId(workflow_version_list[0].workflow_external_id)
+        )
+        listed_by_as_ids = cognite_client.workflows.versions.list(workflow_version_list.as_ids())
 
-        assert len(listed) == len(workflow_version_list)
-        assert listed == workflow_version_list
+        ids_tuples = [tuple(x.__dict__.values()) for x in workflow_version_list.as_ids()]
+        listed_by_tuples = cognite_client.workflows.versions.list(ids_tuples)
 
-    def test_list_workflow_version_limit(
-        self, cognite_client: CogniteClient, workflow_version_list: WorkflowVersionList
-    ) -> None:
-        listed = cognite_client.workflows.versions.list(limit=1)
+        assert (
+            len(workflow_version_list)
+            == len(listed_by_wf_xid)
+            == len(listed_by_wf_version_id)
+            == len(listed_by_as_ids)
+            == len(listed_by_tuples)
+        )
+        assert (
+            workflow_version_list == listed_by_wf_xid == listed_by_wf_version_id == listed_by_as_ids == listed_by_tuples
+        )
 
-        assert len(listed) == 1
+        listed_limit = cognite_client.workflows.versions.list(limit=1)
+        assert len(listed_limit) == 1
 
-    def test_delete_non_existing_raise(self, cognite_client: CogniteClient) -> None:
-        with pytest.raises(CogniteAPIError) as e:
+    def test_delete_non_existing_raise(
+        self, cognite_client: CogniteClient, new_workflow_version: WorkflowVersion
+    ) -> None:
+        with pytest.raises(CogniteAPIError, match="not found"):
             cognite_client.workflows.versions.delete(
-                ("integration_test-non_existing_workflow_version", "1"),
+                [
+                    (new_workflow_version.workflow_external_id, new_workflow_version.version),
+                    (new_workflow_version.workflow_external_id, "non_existing_version"),
+                ],
                 ignore_unknown_ids=False,
             )
+        assert cognite_client.workflows.versions.retrieve(*new_workflow_version.as_id().as_primitive()) is not None
 
-        assert "not found" in str(e.value)
-
-    def test_delete_non_existing(self, cognite_client: CogniteClient) -> None:
+    def test_delete_non_existing(self, cognite_client: CogniteClient, new_workflow_version: WorkflowVersion) -> None:
         cognite_client.workflows.versions.delete(
-            ("integration_test-non_existing_workflow_version", "1"),
+            [
+                (new_workflow_version.workflow_external_id, new_workflow_version.version),
+                (new_workflow_version.workflow_external_id, "non_existing_version"),
+            ],
             ignore_unknown_ids=True,
         )
+        assert cognite_client.workflows.versions.retrieve(*new_workflow_version.as_id().as_primitive()) is None
```

Comment:
I feel like these identifier types should be `namedtuples` to avoid needing an "as primitve" method just to convert it to something we can unpack... but yeah, just my two cents on the existing classes 😓